homeassistant/components/demo/__init__.py

```diff
@@ -269,6 +269,7 @@ async def _insert_statistics(hass):
     # Fake yesterday's temperatures
     metadata = {
         "source": DOMAIN,
+        "name": "Outdoor temperature",
```

Comment:
I suggest that you enable type-checking on line 263:
`async def _insert_statistics(hass: HomeAssistant) -> None:`

Then you will find that you should replace the `metadata` dicts with the corresponding `StatisticMetaData` dataclass:
```python
    metadata = StatisticMetaData(
        source=DOMAIN,
        name="Outdoor temperature",
        statistic_id=f"{DOMAIN}:temperature_outdoor",
        unit_of_measurement="°C",
        has_mean=True,
        has_sum=False,
    )
```